### PR TITLE
Add a general analyzer for postfix generics

### LINKED ISSUE #34: Extend the postfix generics analyzer
The style guide suggests that you should follow the .NET style for generics except for lists, arrays, sequences, options, value options, and reference cells. Right now there is an analyzer for arrays but this can be extended to cover other types as well.

Edit: May also be worth including an analyzer for the .NET style. When postfix is used when it shouldn't be.

https://learn.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#for-types-prefer-prefix-syntax-for-generics-foot-with-some-specific-exceptions

### PR DESCRIPTION
**Pull request by sheridanchris**
This replaces the existing `SquareBracketArrayAnalyzer` and covers every case where the postfix generic style should be applied.

This will close #34

### CODE REVIEW

**Comment by sheridanchris** [comment]
This can easily be modified to create an additional `The generic type is in the postfix position but shouldn't be` diagnostic if necessary. Please let me know and I'll include it!

**Review by dawedawe — APPROVED** [review]
Thanks! Super solid work again.

File: src/Ionide.Analyzers/Style/PostfixGenericsAnalyzer.fs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    fun (context: CliContext) ->
+        async {
+            let ts = ResizeArray<string * range>()
+
+            let collector =
+                { new SyntaxCollectorBase() with
+                    override x.WalkType(t: SynType) =
+                        match t with
+                        | SynType.Array _ -> ts.Add("Prefer postfix syntax for arrays.", t.Range)
+                        | SynType.App(SynType.LongIdent synLongIdent, _, _, _, _, false, _) ->
```
Comment (nojaf) [inline src/Ionide.Analyzers/Style/PostfixGenericsAnalyzer.fs:21]:
```suggestion
                        | SynType.App(typeName = SynType.LongIdent synLongIdent; isPostfix = false) ->
```

**Review by nojaf — APPROVED** [review]
Solid work! Maybe add one more test for a nested scenario.
What gets reported for `let a : array<list<int>> = Array.empty`?

File: src/Ionide.Analyzers/Style/PostfixGenericsAnalyzer.fs (hunk 1 above)
Comment (sheridanchris) [reply src/Ionide.Analyzers/Style/PostfixGenericsAnalyzer.fs]: Done. Thanks!

**Review by nojaf — APPROVED** [review]
Great work! Many thanks!

**Pull request merged**